src/main/java/org/springframework/hateoas/ResourceSupport.java

```diff
@@ -58,14 +65,21 @@ public void add(Link link) {
 	}
 
 	/**
-	 * Adds all given {@link Link}s to the resource.
+	 * Adds all given {@link Link}s or {@link EmbeddedResource}s to the resource.
 	 * 
-	 * @param links
+	 * @param linksOrEmdeddedResource
 	 */
-	public void add(Iterable<Link> links) {
-		Assert.notNull(links, "Given links must not be null!");
-		for (Link candidate : links) {
-			add(candidate);
+	public void add(Iterable linksOrEmdeddedResource) {

```

Comment:
You cannot have both add(Iterable<Link>) and add(Iterable<EmbeddedResource>) as explained here: http://stackoverflow.com/questions/1998544/method-has-the-same-erasure-as-another-method-in-type
Leaving all `add` methods for `Link` and having `addEmbeddeResource` would kind of break a naming convention.
In my opinion it would be better to deprecate `add` in favour of `addLink(s)` and `addEmbeddedResource(s)`. However this would be a change of a public interface of the class, so I thought it was not suitable for a simple PR.